.eslintrc.json

```diff
@@ -0,0 +1,55 @@
+{
+  "root": true,
+  "env": {
+    "browser": true,
+    "es6": true,
+    "jest": true
+  },
+  "overrides": [
+    {
+      "files": ["**/*.tsx", "**/*.js"]
+    }
+  ],
+  "ignorePatterns": ["dist/**/*.js", "dist/**/*.d.ts"],
+  "extends": [
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended",
+    "eslint:recommended",
+    "plugin:react/recommended",
+    "plugin:prettier/recommended",
+    "prettier/react",
+    "prettier/@typescript-eslint"
+  ],
+  "parser": "@typescript-eslint/parser",
+  "parserOptions": {
+    "ecmaFeatures": {
+      "jsx": true
+    },
+    "ecmaVersion": 2018,
+    "sourceType": "module"
+  },
+
+  "settings": {
+    "react": {
+      "version": "detect"
+    }
+  },
+
+  "plugins": ["@typescript-eslint", "react", "prettier", "react-hooks"],
+  "rules": {
+    "prettier/prettier": 1,
+    "@typescript-eslint/explicit-function-return-type": 0,
+    "@typescript-eslint/interface-name-prefix": [1, {"prefixWithI": "always"}],
```

Comment:
I know we use this internally, but IMO its not the best practice for libs.

Reply:
Good call, agreed